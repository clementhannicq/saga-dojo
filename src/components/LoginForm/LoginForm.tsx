import React, { SFC } from "react";

const LoginForm: SFC<any> = ({
  login,
  password,
  error,
  submit,
  loginChange,
  passwordChange
}) => (
  <form
    onSubmit={e => {
      e.preventDefault();
      submit(login, password);
    }}
  >
    {error && <p>Login or password incorrect</p>}
    <input
      type="text"
      value={login}
      onChange={e => loginChange(e.target.value)}
    />
    <input
      type="password"
      value={password}
      onChange={e => passwordChange(e.target.value)}
    />
    <button>Submit</button>
  </form>
);

export default LoginForm as any;
