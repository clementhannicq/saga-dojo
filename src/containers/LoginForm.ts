import { connect } from "react-redux";

import LoginForm from "../components/LoginForm";
import { loginChange, passwordChange, submit } from "./../actions/loginForm";

const mapState = state => ({
  error: state.login.error,
  login: state.login.login,
  password: state.login.password
});

const mapDispatch = {
  loginChange,
  passwordChange,
  submit
};

export default connect(mapState, mapDispatch)(LoginForm);
