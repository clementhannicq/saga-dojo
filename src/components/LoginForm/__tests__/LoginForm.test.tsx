import { shallow } from "enzyme";
import React from "react";
import LoginForm from "../LoginForm";

describe("Components", () => {
  describe("LoginForm", () => {
    it("should render", () => {
      const wrapper = shallow(<LoginForm />);
      expect(wrapper).toMatchSnapshot();
    });
  });
});
