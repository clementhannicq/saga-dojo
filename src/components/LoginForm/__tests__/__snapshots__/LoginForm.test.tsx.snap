// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Components LoginForm should render 1`] = `
<form
  onSubmit={[Function]}
>
  <input
    onChange={[Function]}
    type="text"
  />
  <input
    onChange={[Function]}
    type="password"
  />
  <button>
    Submit
  </button>
</form>
`;
